### PR TITLE
Add RELEASE_CHECKLIST.md file

### PR DESCRIPTION
**Pull request by pmuens**
This file contains a checklist one can go through when releasing a new Serverless version.

### CODE REVIEW

**Comment by eahefnawy** [comment]
This looks like something we'd use internally. I don't see a reason why it should exist in our repo as a markdown file. A google doc that we'd follow maybe better?

I personally have a note in my `eslam` slack channel that outlines the common steps for each release.

File: RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Release checklist
+
+This checklist should be worked through when releasing a new Serverless version.
+
+1. Update Segment.io key (never push the change to GitHub)
+2. Look through all open issues and PRs (if any) and close them / move them to another Milestone
```
Comment (flomotlik) [inline RELEASE_CHECKLIST.md]: I suppose you mean open issues and PRs for the specific milestone?

File: RELEASE_CHECKLIST.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+3. Look through all closed PRs and Issues to see what has changed
```
Comment (flomotlik) [inline RELEASE_CHECKLIST.md]: Same here for the milestone right? There should also be some tools to create a changelog automatically for the future

**Comment by pmuens** [comment]
IMHO a central place where we share the process is important because other might have not idea what to look for when a new release should be published.

Google might be a good place as well.

**Comment by flomotlik** [comment]
Yeah totally agree with @pmuens, documenting this next to the Code makes it clear and the process can easily evolve with the code.

**Comment by pmuens** [comment]
@eahefnawy may you show your checklist so that we can add the missing steps here? Thanks!

**Comment by flomotlik** [comment]
@eahefnawy could you add your thoughts to this so we can merge soon?

**Pull request merged**